network-api/networkapi/wagtailpages/tests.py

```diff
@@ -43,35 +43,42 @@ def setUp(self):
                     hero_image__file__width=1080,
                     hero_image__file__height=720
                 )
+            site = Site.objects.first()
+            site.root_page = homepage
+            site.save()
             # Create the buyersguide page.
             buyersguide = BuyersGuidePage()
             buyersguide.title = 'Privacy not included'
             buyersguide.slug = 'privacynotincluded'
             homepage.add_child(instance=buyersguide)
             buyersguide.save_revision().publish()
 
+            locale = Locale.objects.create(language_code="fr")
+            buyersguide.copy_for_translation(locale, copy_parents=True, alias=True)
+
     def test_homepage(self):
         """
         Test that the homepage works.
         """
         response = self.client.get('/privacynotincluded/')
-        self.assertEqual(response.status_code, 302, 'Homepage should be forwarded to /en/ by default')
+        self.assertEqual(response.status_code, 302)
+        self.assertEqual(
+            response.url,
+            '/en/privacynotincluded/',
+            'Homepage should be forwarded to /en/ by default'
+        )
 
-    @skip("TODO: REENABLE: THIS HAS BEEN TESTED MANUALLY BUT FAILS IN THIS CODE FORM ATM")
     def test_localised_homepage(self):
         """
         Test that the homepage redirects properly under different locale configurations.
         """
-        response = self.client.get('/privacynotincluded/')
-        self.assertEqual(response.status_code, 302, 'simple locale gets redirected')
-
         response = self.client.get('/privacynotincluded', follow=True, HTTP_ACCEPT_LANGUAGE='fr')
         self.assertEqual(
             response.redirect_chain[0][0],
             '/fr/privacynotincluded/',
             'redirects according to HTTP_ACCEPT_LANGUAGE'
         )
-        self.assertEqual(response.status_code, 302)
+        self.assertEqual(response.status_code, 200)
```

Comment:
The point of this test was to make sure that "nonexistent locales" still sends you to a page, so if french now works, we'll need to retain what this test did for a different locale in addition to making sure French passes =)

Reply:
Done :+1:

Reply:
I've copied this test for a valid language that doesn't have a locale, but the `status_code` is still `200` when following the redirect chain, as it ends up at `'/sw/privacynotincluded/'`, which shows the English version of the page.

Reply:
Interesting, I wonder if the behaviour of `redirect_chain` changed across python/wagtail versions O_o